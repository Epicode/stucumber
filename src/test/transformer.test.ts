import Transformer from '../lib/transformer';

describe('transformer', () => {
  it('should transform an AST using given functions', () => {
    const transformer = new Transformer(
      (feature, scenarios) => `describe(${JSON.stringify(feature.name.value)}, () => {${scenarios.join('\n')}});`,
      (feature, scenario, rules) => `it(${JSON.stringify(scenario.name.value)}, () => {${rules.join('\n')}});`,
      (feature, scenario, rule) => `cucumber.rule(${JSON.stringify(rule.value)});` 
    );

    const result = transformer.transform({
      name: {value: 'test feature', location: null},
      scenarios: [
        {
          name: {value: 'test scenario', location: null},
          rules: [
            {value: 'test rule', location: null},
            {value: 'test rule 2', location: null}
          ],
          annotations: []
        }
      ],
      annotations: []
    });

    expect(result).toMatchSnapshot();
  });
});