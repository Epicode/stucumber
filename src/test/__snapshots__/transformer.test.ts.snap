// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformer should transform an AST using given functions 1`] = `
"describe(\\"test feature\\", () => {it(\\"test scenario\\", () => {cucumber.rule(\\"test rule\\");
cucumber.rule(\\"test rule 2\\");});});"
`;
